code/game/objects/items/holy_weapons.dm

```diff
@@ -1,3 +1,177 @@
+// CHAPLAIN CUSTOM ARMORS //
+
+/obj/item/clothing/head/helmet/chaplain
+	name = "crusader helmet"
+	desc = "Deus Vult."
+	icon_state = "knight_templar"
+	item_state = "knight_templar"
+	armor = list("melee" = 41, "bullet" = 15, "laser" = 5,"energy" = 5, "bomb" = 5, "bio" = 2, "rad" = 0, "fire" = 0, "acid" = 50)
+	flags_inv = HIDEMASK|HIDEEARS|HIDEEYES|HIDEFACE|HIDEHAIR
+	flags_cover = HEADCOVERSEYES | HEADCOVERSMOUTH
+	strip_delay = 80
+	dog_fashion = null
+
+/obj/item/clothing/suit/armor/riot/chaplain
+	name = "crusader armour"
+	desc = "God wills it!"
+	icon_state = "knight_templar"
+	item_state = "knight_templar"
+
+/obj/item/holybeacon
+	name = "armaments beacon"
+	desc = "Contains a set of armaments for the chaplain."
+	icon = 'icons/obj/device.dmi'
+	icon_state = "gangtool-red"
+	item_state = "radio"
+	var/used = FALSE
+
+/obj/item/holybeacon/attack_self(mob/user)
+	if(user.mind && (user.mind.isholy) && !used)
+		beacon_armor(user)
+	else
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 60, 1)
+
+/obj/item/holybeacon/proc/beacon_armor(mob/M)
+	if(SSreligion.holy_armor_type)
+		return
+	var/obj/item/storage/box/holy/holy_armor_box
+	var/list/holy_armor_list = typesof(/obj/item/storage/box/holy)
+	var/list/display_names = list()
+	for(var/V in holy_armor_list)
+		var/atom/A = V
+		display_names += initial(A.name)
```

Comment:
I'm not sure how assoc lists work on adding. 

If it's as simple as `+= (initial(A.name) = A)` then epic.

Otherwise 
```
var/the_name = initial(A.name)
display_names += the_name
display_names[then_name] = A
``` 

Also >It only spawns type but you typecast it as atom
...

Reply:
radical centrist over here